Documents/CommitteeSpecificationDrafts/CSD.1/sarif-schema.json

```diff
@@ -1818,9 +1818,7 @@
             }
           }
         }
-      },
-
-      "required": [ "fullyQualifiedLogicalName" ]
```

Comment:
Tempting as it is, we can't retroactively modify CSD.1. We published it to the world. We can only list it as an erratum and correct it in CSD.2.

Reply:
No, the schema is indeed part of the spec.

Reply:
(And if it disagrees with the spec, that is indeed an error, and we capture it in an Erratum.)

Reply:
Here's what the cover page of the spec says:

> **Additional artifacts:**
> This prose specification is one component of a Work Product that also includes:
> - JSON schemas:
>     - sarif-schema.json

Reply:
Right you are!  My bad!  (I'm not sure it makes sense for that to be the case, precisely because it means there's two points of truth--but I can't argue with the spec.  :) )

Reply:
It's unfortunate, but I think it's unavoidable. There must be a prose document for humans to understand; there must be a machine-readable representation that programs can understand, and with today's technology, we don't know how to generate the latter from the former :-)

Reply:
Agree that both should be provided, but couldn't we just provide something like "The prose document is the correct interpretation, the schema is only provided for developer convenience.  In the case where they disagree, the schema is wrong and should/will be corrected."

At which point there's a single point of truth (the doc), we continue providing both, but issues where the schema doesn't agree with the prose doc are easily resolved.

Reply:
It's an interesting discussion. There are arguments both ways:
- English prose can be unclear and unambiguous. A schema is clear and unambiguous. Therefore the schema should be definitive.
BUT!
- English prose can express constraints that a schema _cannot express_. Therefore the spec should be definitive.
I think you have to accept both as definitive and require them to be consistent. In case of conflict, only a human author will know which one was right, and be able to correct the other.

Reply:
I mean, if I'm being argumentative:  you could also argue that if the prose is definitive, if it is ambiguous or could be interpreted in multiple ways by a fluent reader, that's a specification bug.  If we can build the internet on RFCs (with their occasional problems) without schemas or formal specifications of the protocols, we can define a file format without a schema.  Furthermore, if you take the "a schema is clear and unambiguous" to its logical conclusion, we should augment it with an unambiguous formal description of the constraints like uniqueness/the relational constraints/etc.  Only then could you argue that its definitive, because obviously it cannot express the full specification as prose does.

I'll yield to your argument, though, it's pretty compelling, and regardless it's a moot point since the specification says that the schema is part of the spec in CSD.1.